add strimzi-kafka

imjasonh opened the pull request:
strimzi agents: https://github.com/strimzi/strimzi-kafka-operator/blob/a13058f2543aff1bc19c66a12c898b3aa976dd1d/docker-images/kafka-based/kafka/Dockerfile#L73-L78

The intention is for this package + kafka + kafka_exporter to become an image compatible with quay.io/strimzi/kafka

TODO:
- [x] cruise control agent
- [x] third party libs

imjasonh committed:
add strimzi-kafka

Signed-off-by: Jason Hall <jason@chainguard.dev>

EyeCantCU commented on strimzi-kafka.yaml:
```diff
@@ -0,0 +1,60 @@
+      for component in kafka-agent mirror-maker-agent tracing-agent; do
+        cd ./${component}
+        make java_install
+        cp /home/build/.m2/repository/io/strimzi/${component}/${{package.version}}/${component}-${{package.version}}.jar ${{targets.destdir}}/${KAFKA_HOME}/libs/
```
These all get placed under /root so this copy fails. I believe HOME needs to be set to /home/build in the environment to resolve this

imjasonh committed:
updates, add cruise-control

Signed-off-by: Jason Hall <jason@chainguard.dev>

srishtih committed:
Update strimzi-kafka.yaml

srishtih committed:
Update cruise-control.yaml

srishtih committed:
Update cruise-control.yaml

EyeCantCU committed:
fix(strimzi-kafka): Set KAFKA_HOME to /usr/lib/kafka, symlink to /opt/kafka, add runtime deps

This will allow us to use our existing Kafka installation without creating a compat package

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(strimzi-kafka): Remove JARs already provided by Kafka

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(strimzi-kafka): Install scripts for cruise control and kafka

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(strimzi-kafka): Symlink Kafka exporter to /opt/kafka-exporter

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(cruise-control): Remediate CVEs

Remaining CVEs are due to CC building with Kafka's 3.5.x branch

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU commented:
The 2 remaining CVEs are due to Cruise Control using Kafka 3.5.x

EyeCantCU committed:
fix(strimzi-kafka): Add Kafka exporter as runtime dep

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(strimzi-kafka): Clean up duplicate JAR removal

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(strimzi-kafka): Copy over all component JARs

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
chore(cruise-control): Install to libdir

Inline with other Java programs. And symlink to /opt/cruise-control in strimzi-kafka

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>

EyeCantCU committed:
fix(strimzi-kafka): Correct path to Kafka libs

Signed-off-by: RJ Sampson <rj.sampson@chainguard.dev>
